src/cascadia/TerminalCore/Terminal.cpp

```diff
@@ -226,104 +226,68 @@ std::wstring_view Terminal::GetWorkingDirectory() noexcept
 //      nothing to do (the viewportSize is the same as our current size), or an
 //      appropriate HRESULT for failing to resize.
 [[nodiscard]] HRESULT Terminal::UserResize(const til::size viewportSize) noexcept
+try
 {
     const auto oldDimensions = _GetMutableViewport().Dimensions();
     if (viewportSize == oldDimensions)
     {
         return S_FALSE;
     }
 
-    // Shortcut: if we're in the alt buffer, just resize the
-    // alt buffer and put off resizing the main buffer till we switch back. Fortunately, this is easy. We don't need to
-    // worry about the viewport and scrollback at all! The alt buffer never has
-    // any scrollback, so we just need to resize it and presto, we're done.
+    // GH#3494: We don't need to reflow the alt buffer. Apps that use the
+    // alt buffer will redraw themselves. This prevents graphical artifacts.
+    //
+    // This is consistent with VTE
     if (_inAltBuffer())
     {
-        // stash this resize for the future.
+        // _deferredResize will indicate to UseMainScreenBuffer() that it needs to reflow the main buffer.
+        // It's unknown why we defer the reflow to a later time. Performance perhaps?

```

Comment:
Well, hmm. Imagine an application that could set up the alt buffer, resize with dtterm window manipulation, then restore its size before returning to the main buffer. Why bother destructive resizing if the user isn't going to see the intermediate stages?

Reply:
I wrote that comment a long time ago. I realized why we do it in the meantime. My mental model of this was entirely wrong at the time.